### PR TITLE
linuxKernel.kernels: update

### PR DESCRIPTION
**Pull request by lovesegfault**
- linux: 5.15.110 -> 5.15.111
- linux: 6.1.27 -> 6.1.28
- linux: 6.2.14 -> 6.2.15
- linux: 6.3.1 -> 6.3.2
- linux-rt_5_15: 5.15.107-rt62 -> 5.15.111-rt63
- linux-rt_5_4: 5.4.230-rt80 -> 5.4.242-rt81
- linux-rt_6_1: 6.1.19-rt8 -> 6.1.26-rt8
- linux/hardened/patches/4.14: 4.14.313-hardened1 -> 4.14.314-hardened1
- linux/hardened/patches/4.19: 4.19.281-hardened1 -> 4.19.282-hardened1
- linux/hardened/patches/5.10: 5.10.178-hardened1 -> 5.10.179-hardened1
- linux/hardened/patches/5.15: 5.15.108-hardened1 -> 5.15.110-hardened1
- linux/hardened/patches/5.4: 5.4.241-hardened1 -> 5.4.242-hardened1
- linux/hardened/patches/6.1: 6.1.25-hardened1 -> 6.1.27-hardened1

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by roberth** [comment]
`CVE-2023-32233.patch` does not apply in any of the updated versions.

**Comment by fabianhjr** [comment]
Building 6.3.2

**Comment by fabianhjr** [comment]
```
commit f8486683ffa30456e0be4290282a44c4459a3287
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Tue May 2 10:25:24 2023 +0200

    netfilter: nf_tables: deactivate anonymous set from preparation phase
    
    commit c1592a89942e9678f7d9c8030efa777c0d57edab upstream.
    
    Toggle deleted anonymous sets as inactive in the next generation, so
    users cannot perform any update on it. Clear the generation bitmask
    in case the transaction is aborted.
    
    The following KASAN splat shows a set element deletion for a bound
    anonymous set that has been already removed in the same transaction.
```

Present in 6.3.2, needs to be removed from the patches.

**Review by fabianhjr — APPROVED** [review]
LGTM, 6.3.2 builds

**Comment by Ma27** [comment]
Would also test now and merge after that.

File: pkgs/top-level/linux-kernels.nix [hunk 1]
```diff
@@ -158,7 +158,6 @@ in {
         kernelPatches.bridge_stp_helper
         kernelPatches.request_key_helper
         kernelPatches.fix-em-ice-bonding
-        kernelPatches.CVE-2023-32233
```
Comment (Ma27) [inline pkgs/top-level/linux-kernels.nix:161]: Aren't we effectively make the hardened variants vulnerable again? These are AFAICS one patch-level behind upstream.

Comment (fabianhjr) [reply pkgs/top-level/linux-kernels.nix:161]: Yeah, seems so. :S

Comment (Ma27) [reply pkgs/top-level/linux-kernels.nix:161]: Preparing a fix currently.

Comment (Ma27) [reply pkgs/top-level/linux-kernels.nix:161]: Pushed a fix that just adds the patch to the hardened kernels. The other kernels are up-to-date and linux-rt (haven't checked their state) uses its own expression.

**Review by Ma27 — APPROVED** [review]

**Comment by vcunat** [comment]
This broke `nixosTests.nfs3.simple` and thus is blocking `nixos-unstable-small` channel.  Example log: https://hydra.nixos.org/build/219683073

_On Hydra this coincided with `staging-next` merge, but I checked that this test fails on merge commit and succeeds on its parent._

**Comment by K900** [comment]
Reverting https://github.com/torvalds/linux/commit/40882deb83c29d8df4470d4e5e7f137b6acf7ad1 which is the only NFS-related change did not help.

**Comment by vcunat** [comment]
Given that `nixos-unstable` channel is 7 days old now, I'd even consider reverting this whole PR temporarily (unless something better is found today).

**Comment by K900** [comment]
I'd say it's better to disable the test for now, as there are security fixes in these kernels.

**Comment by vcunat** [comment]
Perhaps.  Even if NFS were broken, I'd still expect these updates to be a net benefit for vast majority of "our users".

**Comment by K900** [comment]
Skipped the test: https://github.com/NixOS/nixpkgs/pull/231796

**Comment by Ma27** [comment]
Thx for taking care of that :heart: 

I won't be able to look into that in the upcoming week because I'll be mostly busy with moving to a new flat, but perhaps we're lucky and it fixes itself on the next kernel update :D 

But if it doesn't, I think we should open up an issue here to investigate. After moving is done, I could also try to bisect the kernel and file an upstream report (unless someone is faster).

**Pull request merged**